apps/snitch_core/test/data/model/stock/stock_movement_test.exs

```diff
@@ -0,0 +1,75 @@
+defmodule Snitch.Data.Model.StockMovementTest do
+  @moduledoc false
+
+  use ExUnit.Case, async: true
+  use Snitch.DataCase
+  import Snitch.Factory
+  alias Snitch.Data.Model.StockMovement, as: StockMovementModel
+
+  setup do
+    [stock_item: insert(:stock_item)]
+  end
+
+  describe "create/4" do
+    test "Fails for INVALID stock item id" do
+      assert {:error, changeset} = StockMovementModel.create(1, -1)
+      refute changeset.valid?
+      assert %{stock_item_id: ["does not exist"]} = errors_on(changeset)
+    end
+
+    test "Fails with invalid quantity", context do
+      %{stock_item: stock_item} = context
+      assert {:error, changeset} = StockMovementModel.create("abc", stock_item.id)
+      assert [quantity: {"is invalid", [type: :integer, validation: :cast]}] = changeset
+    end
+
+    test "Inserts with valid attributes", context do
+      %{stock_item: stock_item} = context
+      assert {:ok, stock_movement} = StockMovementModel.create(1, stock_item.id)
+      assert stock_movement.stock_item_id == stock_item.id
+    end
+  end
+
+  describe "get/1" do
```

Comment:
I think we have agreed to **not test** `get`, `get_all`, `delete` if all they do is delegate the work to `QueryHelper`. Please remove these tests. We can probably devise a way to test the QueryHelper methods separately.

Reply:
ok